### PR TITLE
Api 307 remove nested json

### PR DESCRIPTION
**Pull request by tejalcs**
Removed GraphQL and Used Postgres in order to have a flattened structure.

### CODE REVIEW

File: routes/lessons.js [hunk 1]
```diff
@@ -1,91 +1,94 @@
...
 
-  if(!activityId && !moduleId){
+  if(!stageId && !moduleId){
     return res.status(400)
       .send(fetchErrorMessage('default'));
   }
 
-  if(activityId){
-    query = graphQl.generateQuestionQueryByActivityId(activityId);
+  if(stageId){
```
Comment (fahad-cs) [inline routes/lessons.js]: so, if both `stageId` and `moduleId` are provided, `stageId` gets preference?

File: utils/lesson-service-db.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+              'a.name as activity_name, ' +
+              'a.cognito_key as activity_key, ' +
+              'q.id as question_id, ' +
+              'q.question_text as question_name, ' +
+              'q.cognito_key as question_key '+
+              'from question q join activity a on a.id = q.activity_id ' +
+              'join module m on m.id = a.module_id ' +
+              'where a.id=$1';
+  return dbHelper.rawQuery(query, activityId);
+}
```
Comment (fahad-cs) [inline utils/lesson-service-db.js]: Decided to go with raw queries instead of daos, eh? Any particular reason?

File: libs/postgres-helper.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    let queryParams = where? Object.values(where) : {};
+
+    if (where) {
+      query += ` WHERE ${this.constructor.generateWhere(Object.keys(where))}`;
+    }
+
+    return this.db.query(query, queryParams);
+  }
+
+  customSelect(query) {
```
Comment (tejalcs) [inline libs/postgres-helper.js]: No difference, missed removing it.

File: routes/lessons.js (hunk 1 above)
Comment (tejalcs) [inline routes/lessons.js]: Only one of the three will ever be provided. But good point. 👍  I changed the order to have the finest granularity query at the top.

File: utils/lesson-service-db.js (hunk 2 above)
Comment (tejalcs) [inline utils/lesson-service-db.js]: Would have prefered daos. However, all the query's have joins across 3 to 4 tables and so the dynamic select query creation functions wouldn't have worked. Let me know if you have any ideas.

File: routes/lessons.js [hunk 4]
```diff
@@ -38,12 +38,12 @@ router.get('/activities', verifyRequest, function(req, res, next) {
       .send(fetchErrorMessage('default'));
   }
 
-  if(stageId){
-    activitiesQueryFunction = lessonServiceDaos.getActivitiesByStageId;
-    activitiesFilterId = stageId;
-  } else if(moduleId) {
+  if(moduleId){
```
Comment (fahad-cs) [inline routes/lessons.js]: nit: space between `)` and `{`

File: routes/lessons.js [hunk 5]
```diff
@@ -1,91 +1,94 @@
...
+  if(!stageId && !moduleId){
     return res.status(400)
       .send(fetchErrorMessage('default'));
   }
 
-  if(activityId){
-    query = graphQl.generateQuestionQueryByActivityId(activityId);
-  } else if(moduleId) {
-    query = graphQl.generateQuestionQueryByModuleId(moduleId);
+  if(moduleId){
```
Comment (fahad-cs) [inline routes/lessons.js]: nit: space between `)` and `{`

File: routes/lessons.js [hunk 6]
```diff
@@ -1,91 +1,94 @@
...
-  let query;
-  let activityId = req.query.activityId ? parseInt(req.query.activityId) : null;
+router.get('/activities', verifyRequest, function(req, res, next) {
+  let activitiesQueryFunction;
+  let activitiesFilterId;
   let moduleId = req.query.moduleId ? parseInt(req.query.moduleId) : null;
+  let stageId = req.query.stageId ? parseInt(req.query.stageId) : null;
 
-  if(!activityId && !moduleId){
+  if(!stageId && !moduleId){
```
Comment (fahad-cs) [inline routes/lessons.js]: nit: space between `)` and `{`

**Pull request merged**